cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java

```diff
@@ -48,30 +48,30 @@
     private String sqliteVersion = null;
     /**
      * Creates an SQLQueue for the database specified.
-     * @param filename The file where the database is located
+     * @param file The file where the database is located
      * @throws IOException If an problem is encountered creating the DB
      * @throws SQLException If the database cannot be opened.
      */
-    public SQLDatabaseQueue(String filename) throws IOException, SQLException {
-        this(filename, new NullKeyProvider());
+    public SQLDatabaseQueue(File file) throws IOException, SQLException {
+        this(file, new NullKeyProvider());
     }
 
     /**
      * Creates an SQLQueue for the SQLCipher-based database specified.
-     * @param filename The file where the database is located
+     * @param file The file where the database is located
      * @param provider The key provider object that contains the user-defined SQLCipher key.
      *                 Supply a NullKeyProvider to use a non-encrypted database.
      * @throws IOException If a problem occurs creating the database
      * @throws SQLException If the database cannot be opened.
      */
-    public SQLDatabaseQueue(final String filename, KeyProvider provider) throws IOException, SQLException {
+    public SQLDatabaseQueue(final File file, KeyProvider provider) throws IOException, SQLException {
         queue = Executors.newSingleThreadExecutor(new ThreadFactory() {
             @Override
             public Thread newThread(Runnable r) {
-                return new Thread(r, "SQLDatabaseQueue - "+ filename);
+                return new Thread(r, "SQLDatabaseQueue - "+ file);
```

Comment:
The thinking comes from a previous comment by Rich. Previously the datastore names were unique per manager, but now we don't have that restriction. So there is a chance that users could have multiple datastores (documentstores?) open, eg `/a/path/db.sync` and `/b/path/db.sync`. If we follow our existing convention of using the lowest level directory as the name, then the user won't be able to tell the difference.